src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,147 @@
+            System.out.println("  " + tasks[index].getStatusIcon() + " " + tasks[index].getDescription());
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The description of a mark-done cannot be empty.");
+        }
+    }
+
+    public static boolean addTodo(String line, Task[] tasks, int taskNumber) {
+        try {
+            tasks[taskNumber] = new ToDo(line.substring(5));
+            taskNumber++;
+            System.out.println("Got it. I've added this task:");
+            System.out.println("  " + tasks[taskNumber - 1]);
+            System.out.println("Now you have " + taskNumber + " tasks in the list");
+            return true;
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The description of a todo cannot be empty.");
+            return false;
+        }
+    }
+
+    public static boolean addDeadline(String line, Task[] tasks, int taskNumber) {
+        String[] words = line.split(" ");
+        int index = 0;
+        String deadlineDescription = "";
+        String by = "";
+        for (int i = 0; i < words.length; i++) {
+            if (words[i].equals("/by")) {
+                index = i;
+                break;
+            }
+        }
+        if (index == 0) {
+            System.out.println("OOPS!!! The description of a deadline cannot be empty.");
+            return false;
+        }
+        for (int i = 1; i < index; i++) {
+            deadlineDescription = deadlineDescription + words[i] + " ";
+        }
+        for (int i = index + 1; i < words.length; i++) {
+            by = by + words[i] + " ";
+        }
+        tasks[taskNumber] = new Deadline(deadlineDescription, by);
+        taskNumber++;
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks[taskNumber - 1]);
+        System.out.println("Now you have " + taskNumber + " tasks in the list");
+        return true;
+    }
+
+    public static boolean addEvent(String line, Task[] tasks, int taskNumber) {
+        String[] words = line.split(" ");
+        int index = 0;
+        String eventDescription = "";
+        String at = "";
+        for (int i = 0; i < words.length; i++) {
+            if (words[i].equals("/at")) {
+                index = i;
+                break;
+            }
+        }
+        if (index == 0) {
+            System.out.println("OOPS!!! The description of a deadline cannot be empty.");
+            return false;
+        }
+        for (int i = 1; i < index; i++) {
+            eventDescription = eventDescription + words[i] + " ";
+        }
+        for (int i = index + 1; i < words.length; i++) {
+            at = at + words[i] + " ";
+        }
+        tasks[taskNumber] = new Event(eventDescription, at);
+        taskNumber++;
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks[taskNumber - 1]);
```

Comment:
Should the magic literal 1 be declared as a constant instead so that it is easier for you and other programmers to refer to your code in the future? This applies to all magic literals across the code.